contracts/src/v0.8/dev/VRFSubscriptionBalanceMonitor.sol

```diff
@@ -0,0 +1,314 @@
+   */
+  constructor(
+    address linkTokenAddress,
+    address coordinatorAddress,
+    address keeperRegistryAddress,
+    uint256 minWaitPeriodSeconds
+  ) ConfirmedOwner(msg.sender) {
+    setLinkTokenAddress(linkTokenAddress);
+    setVRFCoordinatorV2Address(coordinatorAddress);
+    setKeeperRegistryAddress(keeperRegistryAddress);
+    setMinWaitPeriodSeconds(minWaitPeriodSeconds);
+  }
+
+  /**
+   * @notice Sets the list of subscriptions to watch and their funding parameters.
+   * @param subscriptionIds the list of subscription ids to watch
+   * @param minBalancesJuels the minimum balances for each subscription
+   * @param topUpAmountsJuels the amount to top up each subscription
+   */
+  function setWatchList(
+    uint64[] calldata subscriptionIds,
+    uint96[] calldata minBalancesJuels,
+    uint96[] calldata topUpAmountsJuels
+  ) external onlyOwner {
+    if (subscriptionIds.length != minBalancesJuels.length || subscriptionIds.length != topUpAmountsJuels.length) {
+      revert InvalidWatchList();
+    }
+    uint64[] memory oldWatchList = s_watchList;
+    for (uint256 idx = 0; idx < oldWatchList.length; idx++) {
+      s_targets[oldWatchList[idx]].isActive = false;
+    }
+    for (uint256 idx = 0; idx < subscriptionIds.length; idx++) {
+      if (s_targets[subscriptionIds[idx]].isActive) {
+        revert DuplicateSubcriptionId(subscriptionIds[idx]);
+      }
+      if (subscriptionIds[idx] == 0) {
+        revert InvalidWatchList();
+      }
+      if (topUpAmountsJuels[idx] <= minBalancesJuels[idx]) {
+        revert InvalidWatchList();
+      }
+      s_targets[subscriptionIds[idx]] = Target({
+        isActive: true,
+        minBalanceJuels: minBalancesJuels[idx],
+        topUpAmountJuels: topUpAmountsJuels[idx],
+        lastTopUpTimestamp: 0
+      });
+    }
+    s_watchList = subscriptionIds;
+  }
+
+  /**
+   * @notice Gets a list of subscriptions that are underfunded.
+   * @return list of subscriptions that are underfunded
+   */
+  function getUnderfundedSubscriptions() public view returns (uint64[] memory) {
+    uint64[] memory watchList = s_watchList;
+    uint64[] memory needsFunding = new uint64[](watchList.length);
+    uint256 count = 0;
+    uint256 minWaitPeriod = s_minWaitPeriodSeconds;
+    uint256 contractBalance = LINKTOKEN.balanceOf(address(this));
+    Target memory target;
+    for (uint256 idx = 0; idx < watchList.length; idx++) {
+      target = s_targets[watchList[idx]];
+      (uint96 subscriptionBalance, , , ) = COORDINATOR.getSubscription(watchList[idx]);
+      if (
+        target.lastTopUpTimestamp + minWaitPeriod <= block.timestamp &&
+        contractBalance >= target.topUpAmountJuels &&
+        subscriptionBalance < target.minBalanceJuels
+      ) {
+        needsFunding[count] = watchList[idx];
+        count++;
+        contractBalance -= target.topUpAmountJuels;
+      }
+    }
+    if (count != watchList.length) {
```

Comment:
Would it be more accurate to have this check as `if (count < watchList.length)`? That way it's clear that this code is doing some sort of truncation on the array length.

Reply:
SGTM. No way for count to be greater than.